purchase_requisition_sale_price/models/purchase_requisition.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, fields, api
+from openerp.addons import decimal_precision as dp
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = 'purchase.requisition'
+
+    @api.multi
+    @api.depends('sale_order_id', 'sale_order_id.amount_total',
+                 'sale_order_id.state')
+    def _compute_sale_amount_total(self):
+        for p in self:
+            p.sale_amount_total = 0
+            if p.sale_order_id.state not in ('draft', 'cancel'):
+                p.sale_amount_total = p.sale_order_id.amount_total
+
+    @api.multi
+    @api.depends('line_ids', 'line_ids.total_cost')
+    def _compute_purchase_amount_total(self):
+        for p in self:
+            p.purchase_amount_total = sum(self.line_ids.mapped('total_cost'))
+
+    @api.multi
+    @api.depends('sale_amount_total', 'purchase_amount_total')
+    def _compute_margin(self):
+        for p in self:
+            p.margin = 0
+            if p.sale_amount_total and p.purchase_amount_total:
+                p.margin = 1 - (p.purchase_amount_total / p.sale_amount_total)
```

Comment:
@Daniel-CA ... para eso la linea anterior es una IF:
 if p.sale_amount_total and p.purchase_amount_total: